src/AddressRegistry.Projections.Syndication/BuildingUnit/AddressBuildingUnitLinkProjections.cs

```diff
@@ -0,0 +1,192 @@
+                            addressItem.BuildingUnitPersistentLocalId = buildingUnit.Identificator.ObjectId;
+                            UpdateDbaseRecordField(addressItem, record => record.adresobjid.Value = buildingUnit.Identificator.ObjectId);
+                        }
+                    }
+
+                    context.AddressBulidingUnitLinkExtract.RemoveRange(addressItemsToRemove);
+                }
+            }
+            finally
+            {
+                context.AddressLinkWriterLock.ExitWriteLock();
+            }
+        }
+
+        private async Task<AddressBuildingUnitLinkExtractItem> CreateAddressBuildingUnitLinkExtractItem(AtomEntry<SyndicationItem<Building>> entry, Guid addressId, BuildingUnitSyndicationContent buildingUnit, SyndicationContext context)
+        {
+            var address = await context.AddressLinkAddresses.FindAsync(addressId);
+
+            var dbaseRecord = CreateDbaseRecord(buildingUnit, address, context);
+            return new AddressBuildingUnitLinkExtractItem
+            {
+                AddressId = addressId,
+                BuildingId = entry.Content.Object.Id,
+                BuildingUnitPersistentLocalId = buildingUnit.Identificator.ObjectId,
+                BuildingUnitId = buildingUnit.BuildingUnitId,
+                DbaseRecord = dbaseRecord, //Add address info
+                AddressComplete = address?.IsComplete ?? false,
+                AddressPersistentLocalId = address?.PersistentLocalId,
+            };
+        }
+
+        private byte[] CreateDbaseRecord(BuildingUnitSyndicationContent buildingUnit, AddressLinkSyndicationItem address, SyndicationContext context)
+        {
+            var record = new AddressLinkDbaseRecord
+            {
+                objecttype = { Value = "Gebouweenheid" },
+                adresobjid = { Value = string.IsNullOrEmpty(buildingUnit.Identificator.ObjectId) ? "" : buildingUnit.Identificator.ObjectId },
+            };
+
+            if (address != null)
+            {
+                if (!string.IsNullOrEmpty(address.PersistentLocalId))
+                    record.adresid.Value = Convert.ToInt32(address.PersistentLocalId);
+
+                record.voladres.Value = CreateCompleteAddress(address, context);
+            }
+
+            return record.ToBytes(_encoding);
+        }
+
+        private void UpdateDbaseRecordField(AddressBuildingUnitLinkExtractItem item, Action<AddressLinkDbaseRecord> update)
+        {
+            var record = new AddressLinkDbaseRecord();
+            record.FromBytes(item.DbaseRecord, _encoding);
+            update(record);
+            item.DbaseRecord = record.ToBytes(_encoding);
+        }
+
+        internal static string CreateCompleteAddress(AddressLinkSyndicationItem address, SyndicationContext context)
+        {
+            // update streetname, municipality
+            var streetName = context.StreetNameLatestItems.First(x => x.StreetNameId == address.StreetNameId);
```

Comment:
Not sure if it's possible, but can we add `AsNoTracking()` to these too?

Reply:
true, might increase performance a bit